### PR TITLE
feat: add mastodon

### LINKED ISSUE #3566: announce to mastodon
it is time.

### PR DESCRIPTION
**Pull request by jolheiser**
This PR adds [mastodon](https://joinmastodon.org/) as an announcement pipeline. 🥳 

mastodon-announce

Resolves #3566

### CODE REVIEW

**Review by caarlos0 — APPROVED** [review]
amazing! I'll setup an app to announce goreleaser itself too :D

File: internal/pipe/mastodon/mastodon.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const defaultMessageTemplate = `{{ .ProjectName }} {{ .Tag }} is out! Check it out at {{ .ReleaseURL }}`
+
+type Pipe struct{}
+
+func (Pipe) String() string                 { return "mastodon" }
+func (Pipe) Skip(ctx *context.Context) bool { return !ctx.Config.Announce.Mastodon.Enabled }
+
+type Config struct {
+	Server       string `env:"MASTODON_SERVER,notEmpty"`
```
Comment (jolheiser) [inline internal/pipe/mastodon/mastodon.go:21]: Would you rather have this in the announce config?

Comment (caarlos0) [reply internal/pipe/mastodon/mastodon.go:21]: I think the server could be there yes, maybe only the secrets could be in the env here

Comment (caarlos0) [reply internal/pipe/mastodon/mastodon.go:21]: I'll merge this and do it in a further commit!

**Comment by caarlos0** [comment]
Thanks @jolheiser ! 💚

**Pull request merged**